advanced_source/GuardsOverviewPt1.rst

```diff
@@ -0,0 +1,533 @@
+How TorchDynamo Works - Pt1 - From user code to guards
+======================================================
+
+Welcome
+-------
+
+Welcome to the first entry in a series of technical documents outlining
```

Comment:
I think, here we need to introduce what TorchDynamo so that the users who know nothing about TorchDynamo get a grasp of it. Example:

TorchDynamo is a Python-level just-in-time (JIT) compiler that to helps you accelerate unmodified PyTorch programs. TorchDynamo supports arbitrary PyTorch code, control flow, mutation, and dynamic shapes. Getting started with TorchDynamo is easy — just add a a single line decorator `torch._dynamo.optimize()`.
```suggestion
```